test/SlowTests/Issues/RavenDB_19900.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+using System;
+using System.Threading;
+using Raven.Client.Http;
+using Tests.Infrastructure;
+
+
+namespace SlowTests.Issues
+{
+    public class RavenDB_19900 : ClusterTestBase
+    {
+        public RavenDB_19900(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Fact]
+        public async Task UpdateTopologyTimeoutBehaviourShouldBeAccordingToConfiguration()
+        {
+            var timeout1 = TimeSpan.FromSeconds(20);
+            using (var store = GetDocumentStore())
+            {
+                var executor = store.GetRequestExecutor();
+
+                // delaying response should result in UpdateTopology timeout
+                executor.ForTestingPurposesOnly().DelayRequest = () => { Thread.Sleep(18000); };
+                var e = await Assert.ThrowsAsync<TimeoutException>(async () => await UpdateTopology());
+                Assert.Contains("failed with timeout after 00:00:15", e.ToString());
+
+                // increasing the general timeout should let UpdateTopology() to be executed
+                store.SetRequestTimeout(timeout1);
```

Comment:
using, and close it after UpdateTopology

Reply:
Done